### DISCUSSION TITLE
What are your favorite vanilla emacs commands?

### DISCUSSION THREAD

**Discussion by Esnos33** [meow-edit/meow#695] [General]
Hi, based on issues and discussions on in this github I think a lot of meow users counts that with default meow keybinding they won't need to learn any of the vanilla emacs movements, but I think there are many useful vanilla emacs commands that aren't present in meow.

I this discussion I want everyone to write down their favorite vanilla emacs keybindigs, so new to emacs users that started with meow can better leverage power of emacs, because right now you need to learn both emacs keybindings and meow keybindigs to fully use emacs, which is hard.

I will start and I will probably edit this post many times, each time when I found something that I use from vanilla emacs, in no particular order

1. Isearch
2. move-paragraph
3. set-mark-command
4. forward-paragraph
5. imenu
6. all sexp commands
7. there are meow-query-replace and meow-query-replace-regexp, but they aren't mapped on default keymap.

**Comment by formsandlines** [comment]
For Emacs beginners I would recommend exploring what Emacs offers out of the box first and trying to get familiar with that before learning any 3rd party package (even meow). It really helped me to keep an open mind instead of dismissing things right away because they don’t fit my mental model (yet) or don’t feel familiar. I regretted starting out with Doom Emacs and Evil since I had to disentangle a lot of what they do from Emacs later on when I switched to vanilla Emacs.

Here are the vanilla Emacs commands I added to my meow bindings (with some custom wrappers here and there):

- `undo-only` & `undo-redo` (more common undo/redo behaviour)
- `repeat`
- `open-line`
- `split-line`
- `kill-whole-line`
- `transpose-lines`
- `scroll-up` & `scroll-down`

Other commands I use in meow with the prefix key `/`:

- `end-of-buffer` & `beginning-of-buffer`
- `upcase-dwim` & `downcase-dwim`
- `capitalize-dwim`
- `xref-find-definitions`
- `xref-go-back`
- `xref-find-references`
- `xref-find-apropos`

I find the Isearch and set-mark keybindings quite ergonomic (with control key bound to caps-lock), so I don’t rebind them with meow. However, `meow-search` together with `meow-mark-word` works really well for quick, ad-hoc searching and jumping around in the text.

I remapped movement by paragraph to `{` and `}` using custom commands that call `meow-next-thing` with `'paragraph`, which makes it also select the paragraph similar to `meow-next-word`. It makes movement much faster, so I use that very often.

For sexp editing I mostly use `meow-block` (which lets me select any s-expression and expand the selection outwards) combined with a custom meow state that uses commands from the symex package for more fine-grained structural editing.

**Comment by okamsn** [comment]
When using more limited editors, I quickly notice the lack of a feature like `fill-paragraph` (`M-q`). It's not rewrapping lines, it's reshaping a paragraph in a way that makes sense for the given context. For example, in Org mode, it's aware of lists and tables.

**Comment by gs-101** [comment]
I agree with the examples presented here, but I'll also add `align` and `align-regexp` to the list.

You can even just give it one character and they align everything correctly.
